wrangler-transform/src/main/java/io/cdap/wrangler/Precondition.java

```diff
@@ -88,4 +98,32 @@ public boolean apply(Row row) throws PreconditionException {
       }
     }
   }
+
+  @Override
+  public Relation transform(RelationalTranformContext relationalTranformContext, Relation relation) {
+    if (!isConditionSQL) {
+      return new InvalidRelation("Condition is not specified as SQL");
+    }
+
+    Schema inputSchema = getSchema();
+
+    ExpressionConversionContext context = new ExpressionConversionContext();
+    SQLDialect dest = isPushdownEnabled() ? SQLDialect.BIGQUERYSQL : SQLDialect.SPARKSQL;
+    ExpressionFactory<String> expressionFactory = context.getConversionFactory(
+            SQLDialect.POSTGRESQL, dest, inputSchema);
+
+    Expression compiledExpression = expressionFactory.compile(condition);
+
+    return relation.filter(compiledExpression);
+  }
+
+  // TODO: Find out from where we can get schema information. Perhaps RelationalTranformContext
+  private Schema getSchema() {
+    return null;
+  }
+
+  // TODO: Find how this info can be obtained. Perhaps RelationalTranformContext
+  boolean isPushdownEnabled() {
```

Comment:
Pushdown will always be enabled. The Spark engine is also a pushdown engine, so the answer will always be yes.

The dialect needs to be selected by the SQL Engine implementation.